mongodb/util/helpers.go

```diff
@@ -25,3 +35,39 @@ func LogNonNilError(err error) {
 		log.Printf("Unexpected error: %v", err)
 	}
 }
+
+// IsValidEmailAddress returns true if the passed string is a valid email address
+func IsValidEmailAddress(email string) bool {
+	return reValidEmail.MatchString(email)
+}
+
+// TryExtractFirstLastNameAndEmail attempts to extract first, last name and email address
+func TryExtractFirstLastNameAndEmail(contents string) (firstName string, lastName string, emailAddress string) {
+	if !IsValidEmailAddress(contents) {
+		// if the data string is not an email, set first/last name to the passed contents
+		firstName = contents
+		lastName = contents
+		emailAddress = ""
+		return
+	}
+	emailAddress = contents
+
+	part := strings.Split(contents, "@")[0]
+	tentative := strings.SplitN(part, ".", 2)
+	if len(tentative) < 2 {
+		// if the contents does not contain a dot, attempt to split by underscore
+		tentative = strings.SplitN(part, "_", 2)
+	}
+
+	if len(tentative) < 2 {
+		// could not find 2 parts, set both as the same
+		firstName = tentative[0]
+		lastName = tentative[0]
+		emailAddress = contents
```

Comment:
can remove this line - already set on line 53

Reply:
good catch - the logic was bad so I fixed it.